unit_tests/utilities/test_zaza_utilities_swift.py

```diff
@@ -91,12 +91,21 @@ def test_get_swift_storage_topology(self):
         self.patch_object(juju_utils, 'get_full_juju_status')
         self.patch_object(zaza.model, 'get_application_config')
         self.patch_object(zaza.model, 'get_units')
+        self.patch_object(zaza.model, 'get_unit_public_address')
+
+        def _get_unit_public_address(u):
+            return u.public_address
+
+        self.get_unit_public_address.side_effect = _get_unit_public_address
+
         juju_status = mock.MagicMock()
         juju_status.applications = {}
         self.get_full_juju_status.return_value = juju_status
 
         for app_name, units in app_units.items():
-            expected_topology[units[0].public_address]['unit'] = units[0]
+            # ip = zaza.model.get_unit_public_address(units[0])
+            ip = units[0].public_address
```

Comment:
shouldn't it be the previous line instead, i.e. shouldn't we call get_unit_public_address() instead of using the public_address property?

Reply:
Hmm, strange.  Good catch; I will check the code again.

Reply:
So, the commented out line is incorrect. This is actually a unit test of a test (yes, really!) and so *here* units[0].public_address is the mock further up in the code.